src/cargo/core/resolver/mod.rs

```diff
@@ -256,12 +257,42 @@ impl<'a> Iterator for DepsNotReplaced<'a> {
     }
 }
 
+enum RcList<T> {
+    Data(Rc<(T, RcList<T>)>),
+    Empty,
+}
```

Comment:
I guess `Rc<LinkedList<T>>` would get us what we wanted. Though that's kind of questionable at best...